packages/tpetra/core/ext/TpetraExt_MatrixMatrix_def.hpp

```diff
@@ -1178,135 +1175,53 @@ void Add(
-           const Values Bvals_, const typename RowPtrs::const_type Browptrs_, const ColInds Bcolinds_, const Scalar scalarB_,
-           Values Cvals_, const RowPtrs Crowptrs_, ColInds Ccolinds_) :
-    Avals(Avals_), Arowptrs(Arowptrs_), Acolinds(Acolinds_), scalarA(scalarA_),
-    Bvals(Bvals_), Browptrs(Browptrs_), Bcolinds(Bcolinds_), scalarB(scalarB_),
-    Cvals(Cvals_), Crowptrs(Crowptrs_), Ccolinds(Ccolinds_) {}
-  KOKKOS_INLINE_FUNCTION void operator()(const size_t i) const
-  {
-    //get views of row i in A and B
-    auto Arowstart = Arowptrs(i);
-    auto Arowlen = Arowptrs(i + 1) - Arowstart;
-    auto Browstart = Browptrs(i);
-    auto Browlen = Browptrs(i + 1) - Browstart;
-    Ordinal CrowStart = Crowptrs(i);
-    Ordinal ai = 0;
-    Ordinal bi = 0;
-    Ordinal numInserted = 0;
-
-    // Ordinal and the type of Arowlen (and Browlen) might not have
-    // the same sign.  In order to avoid build warnings about signed /
-    // unsigned comparisons at the while loop below, we cast to the
-    // "common type" of the two values before comparing them.
-    typedef typename std::common_type<decltype (Arowlen), Ordinal>::type
-      common_type;
-    while (static_cast<common_type> (ai) < static_cast<common_type> (Arowlen) &&
-           static_cast<common_type> (bi) < static_cast<common_type> (Browlen))
-    {
-      Ordinal Acol = Acolinds(Arowstart + ai);
-      Ordinal Bcol = Bcolinds(Browstart + bi);
-      //next col in C is minimum of two next cols in A and B
-      Ordinal Ccol = Acol < Bcol ? Acol : Bcol;
-      Scalar val = 0;
-      if(Acol == Ccol)
-      {
-        val += scalarA * Avals(Arowstart + ai);
-        ai++;
-      }
-      if(Bcol == Ccol)
-      {
-        val += scalarB * Bvals(Browstart + bi);
-        bi++;
-      }
-      Cvals(CrowStart + numInserted) = val;
-      Ccolinds(CrowStart + numInserted) = Ccol;
-      numInserted++;
-    }
-    for(; static_cast<common_type> (ai) < static_cast<common_type> (Arowlen); ai++)
-    {
-      Cvals(CrowStart + numInserted) = scalarA * Avals(Arowstart + ai);
-      Ccolinds(CrowStart + numInserted) = Acolinds(Arowstart + ai);
-      numInserted++;
-    }
-    for(; static_cast<common_type> (bi) < static_cast<common_type> (Browlen); bi++)
-    {
-      Cvals(CrowStart + numInserted) = scalarB * Bvals(Browstart + bi);
-      Ccolinds(CrowStart + numInserted) = Bcolinds(Browstart + bi);
-      numInserted++;
-    }
-  }
-  const Values Avals;
-  const typename RowPtrs::const_type Arowptrs;
-  const ColInds Acolinds;
-  const Scalar scalarA;
-  const Values Bvals;
-  const typename RowPtrs::const_type Browptrs;
-  const ColInds Bcolinds;
-  const Scalar scalarB;
-  Values Cvals;
-  const RowPtrs Crowptrs;
-  ColInds Ccolinds;
-};
+  using Teuchos::TimeMonitor;
+  TEUCHOS_TEST_FOR_EXCEPTION(Arowptrs.dimension_0() != Browptrs.dimension_0(), std::runtime_error, "Can't add matrices with different numbers of rows.");
```

Comment:
Is this a new test?

Reply:
or was it there before?

Reply:
It was there before. Git thinks it's new because the indentation changed for this function.